src/policy/vmspace.rs

```diff
@@ -0,0 +1,233 @@
+                worker: GCWorkerMutRef,
+            ) -> ObjectReference;
+        }
+    }
+}
+
+impl<VM: VMBinding> Space<VM> for VMSpace<VM> {
+    fn as_space(&self) -> &dyn Space<VM> {
+        self
+    }
+    fn as_sft(&self) -> &(dyn SFT + Sync + 'static) {
+        self
+    }
+    fn get_page_resource(&self) -> &dyn PageResource<VM> {
+        self.space().get_page_resource()
+    }
+    fn common(&self) -> &CommonSpace<VM> {
+        self.space().common()
+    }
+
+    fn initialize_sft(&self) {
+        if self.inner.is_some() {
+            self.common().initialize_sft(self.as_sft())
+        }
+    }
+
+    fn release_multiple_pages(&mut self, _start: Address) {
+        panic!("immortalspace only releases pages enmasse")
+    }
+
+    fn verify_side_metadata_sanity(&self, side_metadata_sanity_checker: &mut SideMetadataSanity) {
+        side_metadata_sanity_checker.verify_metadata_context(
+            std::any::type_name::<Self>(),
+            &SideMetadataContext {
+                global: self.args.global_side_metadata_specs.clone(),
+                local: vec![],
+            },
+        )
+    }
+
+    fn address_in_space(&self, start: Address) -> bool {
+        if let Some(space) = self.space_maybe() {
+            space.address_in_space(start)
+        } else {
+            false
+        }
+    }
+}
+
+use crate::scheduler::GCWorker;
+use crate::util::copy::CopySemantics;
+
+impl<VM: VMBinding> crate::policy::gc_work::PolicyTraceObject<VM> for VMSpace<VM> {
+    fn trace_object<Q: ObjectQueue, const KIND: crate::policy::gc_work::TraceKind>(
+        &self,
+        queue: &mut Q,
+        object: ObjectReference,
+        _copy: Option<CopySemantics>,
+        _worker: &mut GCWorker<VM>,
+    ) -> ObjectReference {
+        self.trace_object(queue, object)
+    }
+    fn may_move_objects<const KIND: crate::policy::gc_work::TraceKind>() -> bool {
+        false
+    }
+}
+
+impl<VM: VMBinding> VMSpace<VM> {
+    pub fn new(args: &mut CreateSpecificPlanArgs<VM>) -> Self {
+        let args_clone = CreateSpecificPlanArgs {
+            global_args: CreateGeneralPlanArgs {
+                vm_map: args.global_args.vm_map,
+                mmapper: args.global_args.mmapper,
+                heap: HeapMeta::new(), // we do not use this
+                options: args.global_args.options.clone(),
+                gc_trigger: args.global_args.gc_trigger.clone(),
+                scheduler: args.global_args.scheduler.clone(),
+            },
+            constraints: args.constraints,
+            global_side_metadata_specs: args.global_side_metadata_specs.clone(),
+        };
+        if !args.global_args.options.vm_space_start.is_zero() {
+            Self {
+                inner: Some(Self::create_space(args, None)),
+                args: args_clone,
+            }
+        } else {
+            Self {
+                inner: None,
+                args: args_clone,
+            }
+        }
```

Comment:
If we want to further reduce repetition, we can use the `bool::then` method.  But since the `Self` struct is not very big, the existing code looks pretty readable.  

```suggestion
        let inner = (!args.global_args.options.vm_space_start.is_zero()).then(|| {
            Self::create_space(args, None)
        });
        Self {
            inner,
            args: args_clone,
        }
```

Reply:
Done